### PR TITLE
todolist-1102:created sign up , login API's

### PR DESCRIPTION
**Pull request by 04vignesh45**

### CODE REVIEW

File: driver/driver.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+	//third party package
+	
+
+	_ "github.com/lib/pq"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+var err error
```
Comment (brocklesnarvj) [inline driver/driver.go]: try to avoid global variable

File: handler/handler.go [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+		})
+	}
+	// jobId := strconv.Itoa(int(post.User_id))
+	// _,err:= repository.GetUserId(jobId)
+	// if err != nil {
+	// 	log.Error("Error:'User ID not found' status:404")
+	// 	return c.Status(http.StatusNotFound).JSON(map[string]interface{}{
+	// 		"Error":  "User ID not found",
+	// 		"status": 404,
+	// 	})
```
Comment (brocklesnarvj) [inline handler/handler.go]: Remove Unwanted comments before commit and push

File: models/models.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	User_id     uint   `json:"-"                     gorm:"primaryKey;autoIncrement:true"`
+	Email       string `json:"email"                 gorm:"column:email;type:varchar(35)"`
+	Username    string `json:"username"              gorm:"column:user_name;type:varchar(100)"`
+	Password    string `json:"password"              gorm:"column:password;type:varchar(100)"`
+	PhoneNumber string `json:"phone_number"          gorm:"column:phone_number;type:varchar(15)"`
+}
+
+type TaskDetails struct {
+	TASK_ID   uint   `json:"-"                   gorm:"primaryKey;autoIncrement:true"`
+	// User_id   uint   `json:"-"                   gorm:"type:bigint references Information(user_id)"`
```
Comment (brocklesnarvj) [inline models/models.go]: remove unused variable don't comment it

File: repository/repository.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+import (
+	//user defined package
+
+	"todo/models"
+
+	"gorm.io/gorm"
+)
+
+var Db *gorm.DB
```
Comment (brocklesnarvj) [inline repository/repository.go]: Avoid Global variable

File: go.sum
Comment (brocklesnarvj) [inline go.sum:1]: 1) Avoid Global Variable 
2) Remove Unused variables and package don't comment on it.

File: driver/driver.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	user := os.Getenv("USER")
+
+	//connecting to postgres-database
+	connection := fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=disable", host, port, user, password, dbname)
+	repository.Db,err= gorm.Open(postgres.Open(connection), &gorm.Config{})
+	if err != nil {
+	
+		fmt.Println("error in connecting with database")
+	}
+
```
Comment (Niru2k) [inline driver/driver.go]: When the DB connection will close?

File: handler/handler.go [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+			"Error":  "Username field should not be empty",
+			"status": 400,
+		})
+	}
+	//password should have minimum 8 character
+	if len(user.Password) < 8 {
+		log.Error("error:'Password should be more than 8 characters' status:400")
+		return c.Status(http.StatusBadRequest).JSON(map[string]interface{}{
+			"Error":  "Password should be more than 8 characters",
+			"status": 400,
```
Comment (Niru2k) [inline handler/handler.go]: For wrong password use status code "403" which means Forbidden.

File: handler/handler.go [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+	})
+}
+
+// Login API
+func Login(c *fiber.Ctx) error {
+	log := logs.Logs()
+	log.Info("login api called successfully")
+	var login models.Information
+	if err := c.BodyParser(&login); err != nil {
+		log.Error("error:'Invalid Format' status:400")
```
Comment (Niru2k) [inline handler/handler.go:118]: It is internal server error, Use status code "500"

File: handler/handler.go [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+		return c.Status(http.StatusBadRequest).JSON(map[string]interface{}{
+			"error":  "Invalid Format",
+			"status": 400,
```
Comment (Niru2k) [inline handler/handler.go:128]: Here also use "500"

File: handler/handler.go [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+		})
+	}
+	//verify the email whether its already registered in the SignUp API or not
+	verify, err := repository.ReadUserByEmail(login)
+	if err == nil {
+		//checks whether the given password matches with the email
+		if err := bcrypt.CompareHashAndPassword([]byte(verify.Password), []byte(login.Password)); err != nil {
+			log.Error("error:'Password Not Matching' status:400")
+			return c.Status(http.StatusBadRequest).JSON(map[string]interface{}{
+				"Error":  " Password Not Matching",
```
Comment (Niru2k) [inline handler/handler.go:138]: For wrong password use status code "403" which means Forbidden.

File: handler/handler.go [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+
+// Task Posting API
+func TaskRemainder(c *fiber.Ctx) error {
+	log := logs.Logs()
+	log.Info(" TaskRemainder api called successfully")
+	var post models.TaskDetails
+	if err := c.BodyParser(&post); err != nil {
+		log.Error("error:'invalid format' status:400")
+		return c.Status(http.StatusBadRequest).JSON(map[string]interface{}{
+			"Error":  "invalid format",
```
Comment (Niru2k) [inline handler/handler.go:185]: This internal server error not bad request

**Review by Niru2k — COMMENTED** [review]
Reviewed

**Review by brocklesnarvj — APPROVED** [review]

**Pull request merged**